packages/MainUI/components/Toolbar.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { Box } from '@mui/material';
+import { useToolbar } from '@workspaceui/etendohookbinder/src/hooks/useToolbar';
+import PlusIcon from '@workspaceui/componentlibrary/src/assets/icons/plus.svg';
+import SaveIcon from '@workspaceui/componentlibrary/src/assets/icons/save.svg';
+import SearchIcon from '@workspaceui/componentlibrary/src/assets/icons/search.svg';
+import RefreshIcon from '@workspaceui/componentlibrary/src/assets/icons/refresh-cw.svg';
+import TrashIcon from '@workspaceui/componentlibrary/src/assets/icons/trash-2.svg';
+import GridIcon from '@workspaceui/componentlibrary/src/assets/icons/grid.svg';
+import DownloadIcon from '@workspaceui/componentlibrary/src/assets/icons/download.svg';
+import PaperclipIcon from '@workspaceui/componentlibrary/src/assets/icons/paperclip.svg';
+import TopToolbar from '@workspaceui/componentlibrary/src/components/Table/Toolbar';
+import { theme } from '@workspaceui/componentlibrary/src/theme';
+import { IconName, IconSize, ToolbarResponse, ToolbarProps, ButtonConfig } from './types';
+
+const iconMap: Record<IconName, React.FC<unknown>> = {
+  plus: PlusIcon,
+  save: SaveIcon,
+  trash: TrashIcon,
+  'refresh-cw': RefreshIcon,
+  search: SearchIcon,
+  grid: GridIcon,
+  download: DownloadIcon,
+  paperclip: PaperclipIcon,
+};
+
+export const Toolbar: React.FC<ToolbarProps> = ({ windowId, tabId }) => {
+  const { toolbar, loading } = useToolbar(windowId, tabId);
+
+  if (loading) {
+    return (
+      <Box display="flex" justifyContent="center" alignItems="center" height={64}>
+        Loading...
+      </Box>
+    );
+  }
+
+  const createToolbarConfig = () => {
+    const buttons = (toolbar as ToolbarResponse)?.response?.buttons || [];
+
+    return {
+      leftSection: {
+        buttons: buttons
+          .filter(btn => ['NEW', 'REFRESH', 'SAVE'].includes(btn.id))
+          .map(btn => {
+            const buttonConfig: ButtonConfig = {
+              key: btn.id,
+              icon: React.createElement(iconMap[btn.icon]),
+              tooltip: btn.name,
+              onClick: () => {},
+              disabled: !btn.enabled,
+              height: IconSize,
+              width: IconSize,
+              fill: theme.palette.baselineColor.neutral[0],
+            };
+
+            if (btn.id === 'NEW') {
+              buttonConfig.iconText = btn.name;
+            }
+
+            buttonConfig['sx'] = (() => {
+              if (btn.id === 'NEW') {
```

Comment:
In these cases @Sankku, I think it's better move such words from enumerated type to constants

Reply:
👍🏼